From 40d80e3e31eb2f9f1bf55861b84c2df5b2051957 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Tue, 6 Nov 2018 13:35:40 +0200
Subject: [PATCH 5059/5242] MLK-20242: drm/imx/dcss: fix brightness for
 REC.709

commit  1e3fcb39a6fca44bae78a12b50c7112dc95e96ae from
https://source.codeaurora.org/external/imx/linux-imx.git

The brightness, when REC709 was used in the configuration of the pipes,
was lower than expected.

The reason was the HDR10 configuration application that was used to
create the tables had a parameter that was wrongly set.

The tables were re-generated with the proper setting.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-hdr10-tables.h |  173 +++---------------------------
 1 file changed, 17 insertions(+), 156 deletions(-)

diff --git a/drivers/gpu/imx/dcss/dcss-hdr10-tables.h b/drivers/gpu/imx/dcss/dcss-hdr10-tables.h
index ee18208..edf9fe1 100644
--- a/drivers/gpu/imx/dcss/dcss-hdr10-tables.h
+++ b/drivers/gpu/imx/dcss/dcss-hdr10-tables.h
@@ -1390,39 +1390,7 @@
 	0xfffffe00, 0xfffffe00, 0x0, 0xfffffe00, 0xfffffe00, 0x3ff, 0x1ff,
 	0x1ff, 0xe, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff,
 	/* table descriptor */
-	0xfff17664, 0x0,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x357d, 0x1c13, 0x3b1, 0x5e4, 0x4e65, 0xf8, 0x166, 0x781, 0x4c5b,
-	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0xfff27664, 0x0,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x37d8, 0x19c3, 0x3a6, 0x626, 0x4e26, 0xf5, 0x176, 0x858, 0x4b74,
-	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0xfff47664, 0x0,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x3403, 0x1d94, 0x3aa, 0x719, 0x4b7d, 0x2ab, 0x15b, 0x6f0, 0x4cf7,
-	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0xfff87664, 0x0,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x4ae5, 0x68c, 0x3d1, 0x8f4, 0x4b4d, 0x100, 0x2b9, 0x3a3, 0x4ee6,
-	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0xfff176a4, 0x0,
+	0xfff176e4, 0x0,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1430,7 +1398,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x1, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0xfff276a4, 0x0,
+	0xfff276e4, 0x0,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1438,7 +1406,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x1, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0xfff476a4, 0x0,
+	0xfff476e4, 0x0,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1446,7 +1414,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x1, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0xfff876a4, 0x0,
+	0xfff876e4, 0x0,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1454,32 +1422,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x1, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff27664, 0x1,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x5903, 0xfffffc3f, 0x0, 0x0, 0x5541, 0x0, 0x0, 0x101, 0x5440,
-	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0x7ff47664, 0x1,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x5218, 0x440, 0xfffffee9, 0x18e, 0x51c8, 0x1ec, 0xffffffdc,
-	0xffffffa1, 0x55c4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff,
-	0x3fff, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
-	0xffc0000,
-	/* table descriptor */
-	0x7ff87664, 0x1,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x76e7, 0xffffde5b, 0x0, 0xcb, 0x5476, 0x0, 0xc9, 0xfffffc60,
-	0x5818, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0x7ff276a4, 0x1,
+	0x7ff276e4, 0x1,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1487,7 +1430,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff476a4, 0x1,
+	0x7ff476e4, 0x1,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1496,7 +1439,7 @@
 	0x3fff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
 	0xffc0000,
 	/* table descriptor */
-	0x7ff876a4, 0x1,
+	0x7ff876e4, 0x1,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1504,32 +1447,7 @@
 	0x4212, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff17664, 0x2,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x51a9, 0x399, 0x0, 0x0, 0x5541, 0x0, 0x0, 0xfffffefc, 0x5646,
-	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0x7ff47664, 0x2,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x4eb2, 0x785, 0xffffff0a, 0x18e, 0x51c8, 0x1ec, 0xffffffd7,
-	0xfffffea6, 0x56c4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff,
-	0x3fff, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
-	0xffc0000,
-	/* table descriptor */
-	0x7ff87664, 0x2,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x71eb, 0xffffe356, 0x0, 0xcb, 0x5476, 0x0, 0xc9, 0xfffffb53,
-	0x5925, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0x7ff176a4, 0x2,
+	0x7ff176e4, 0x2,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1537,7 +1455,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff476a4, 0x2,
+	0x7ff476e4, 0x2,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1546,7 +1464,7 @@
 	0x3fff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
 	0xffc0000,
 	/* table descriptor */
-	0x7ff876a4, 0x2,
+	0x7ff876e4, 0x2,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1554,32 +1472,7 @@
 	0x42dc, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff17664, 0x4,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x58a0, 0xfffffb67, 0x13a, 0xfffffe50, 0x58f5, 0xfffffdfc, 0x24,
-	0x60, 0x54bd, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
-	0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0x7ff27664, 0x4,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x5c88, 0xfffff783, 0x137, 0xfffffe3d, 0x5902, 0xfffffe02, 0x25,
-	0x15f, 0x53be, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
-	0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0x7ff87664, 0x4,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x7b92, 0xffffd86a, 0x145, 0xfffffe75, 0x58e1, 0xfffffdeb, 0xfb,
-	0xfffffcb7, 0x5790, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff,
-	0x3fff, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
-	0xffc0000,
-	/* table descriptor */
-	0x7ff176a4, 0x4,
+	0x7ff176e4, 0x4,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1587,7 +1480,7 @@
 	0x48, 0x3f8e, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff276a4, 0x4,
+	0x7ff276e4, 0x4,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1595,7 +1488,7 @@
 	0x107, 0x3ece, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff876a4, 0x4,
+	0x7ff876e4, 0x4,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1604,38 +1497,6 @@
 	0x3fff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
 	0xffc0000,
 	/* table descriptor */
-	0x7ff17664, 0x8,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x3cf8, 0x1849, 0x0, 0xffffff6d, 0x55d4, 0x0, 0xffffff6f, 0x351,
-	0x5282, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0x7ff27664, 0x8,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x3fa7, 0x159a, 0x0, 0xffffff67, 0x55da, 0x0, 0xffffff68, 0x450,
-	0x5189, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0x7ff47664, 0x8,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x3b26, 0x1a56, 0xffffffc5, 0x103, 0x524d, 0x1f1, 0xffffff60,
-	0x2cb, 0x5316, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
-	0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
-	0x7fff7664, 0xf,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x5541, 0x0, 0x0, 0x0, 0x5541, 0x0, 0x0, 0x0, 0x5541, 0x0, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0, 0x0, 0x0,
-	0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
-	/* table descriptor */
 	0xfff0ffe4, 0xf,
 	/* table length */
 	0x1d,
@@ -1644,7 +1505,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
 	0x0, 0x0,
 	/* table descriptor */
-	0x7ff176a4, 0x8,
+	0x7ff176e4, 0x8,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1652,7 +1513,7 @@
 	0x3de1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff276a4, 0x8,
+	0x7ff276e4, 0x8,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1660,7 +1521,7 @@
 	0x3d27, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff476a4, 0x8,
+	0x7ff476e4, 0x8,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1668,7 +1529,7 @@
 	0x431, 0x7ca0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
 	0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7fff76a4, 0xf,
+	0x7fff76e4, 0xf,
 	/* table length */
 	0x1d,
 	/* table data */
-- 
1.7.9.5

